src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/GenericLookupResult.cs

```diff
@@ -481,10 +487,7 @@ public override ISymbolNode GetTarget(NodeFactory factory, Instantiation typeIns
             if (factory.Target.Abi == TargetAbi.CoreRT)
             {
                 MethodDesc instantiatedMethod = _method.InstantiateSignature(typeInstantiation, methodInstantiation);
-
-                // https://github.com/dotnet/corert/issues/2342 - we put a pointer to the virtual call helper into the dictionary
-                // but this should be something that will let us compute the target of the dipatch (e.g. interface dispatch cell).
-                return factory.ReadyToRunHelper(ReadyToRunHelperId.VirtualCall, instantiatedMethod);
+                return factory.InterfaceDispatchCell(instantiatedMethod);
             }
```

Comment:
Because there's still a small delta (the "callsite ID" of the cell) that we do differently (by differently I mean "don't populate") that makes this not align right).

I could fix that, but I figured this change is already bigger than anyone would want to review and these dictionary differences are still on our radar to look at.